pkg/api/v1beta2/defaults.go

```diff
@@ -95,7 +97,43 @@ func init() {
 		},
 		func(obj *Endpoints) {
 			if obj.Protocol == "" {
-				obj.Protocol = "TCP"
+				obj.Protocol = ProtocolTCP
```

Comment:
in general we want each API version to be siloed so they can evolve independently.